addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,138 @@
+	<bridge-type id="account">
+		<label>iCloud Account</label>
+		<description>Provides various information for iCloud devices registered with this account.</description>
+
+		<channels>
+			<channel id="owner" typeId="Owner" />
+			<channel id="numberOfDevices" typeId="NumberOfDevices" />
+			<channel id="refresh" typeId="Refresh" />
+		</channels>
+
+		<config-description>
+			<parameter name="appleId" type="text" required="true">
+				<context>email</context>
+				<label>Apple Id</label>
+				<description>Apple Id (e-mail) to access iCloud information.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Apple iCloud password for the given Id.</description>
+			</parameter>
+			<parameter name="refreshTimeInMinutes" type="integer" min="5" max="65535" unit="min">
+				<label>Refresh</label>
+				<description>Refresh time in minutes.</description>
+				<default>5</default>
+			</parameter>
+			<parameter name="googleAPIKey" type="text">
+				<label>Google API Key</label>
+				<description>Google API key to be used for address lookup calls (to translate coordinates into a human readable string). Without a key the number of daily request are limited and the bindings address lookup functionality will stop working.</description>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>iCloud Device</label>
+
+		<channels>
+			<channel typeId="BatteryStatus" id="batteryStatus" />
+			<channel typeId="system.battery-level" id="batteryLevel" />
+			<channel typeId="FindMyPhone" id="findMyPhone" />
+			<channel typeId="Location" id="location" />
+			<channel typeId="LocationAccuracy" id="locationAccuracy" />
+			<channel typeId="LocationLastUpdate" id="locationLastUpdate" />
+			<channel typeId="DistanceFromHome" id="distanceFromHome" />
+			<channel typeId="AddressStreet" id="addressStreet" />
+			<channel typeId="AddressCity" id="addressCity" />
+			<channel typeId="AddressCountry" id="addressCountry" />
+			<channel typeId="FormattedAddress" id="formattedAddress" />
+			<channel typeId="DeviceName" id="deviceName" />
+		</channels>
+
+		<representation-property>deviceId</representation-property>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device id</label>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="NumberOfDevices">
+		<item-type>Number</item-type>
+		<label>Number of devices</label>
+		<description>Number of registered devices with this iCloud account.</description>
+	</channel-type>
+	<channel-type id="Owner">
+		<item-type>String</item-type>
+		<label>Owner</label>
+		<description>Registered owner of this iCloud account.</description>
+	</channel-type>
+	<channel-type id="Refresh">

```

Comment:
This is a bit of a weird feature and can lead to confusion. We have "REFRESH" commands for channels in order to get recent data - imho that should suffice here as well and this channel should be removed.

Reply:
@kaikreuzer the channel serves a valuable purpose by updating all channels from the iCloud. Before removing it the alternative should clearly be defined. Could you explain your approach?

Reply:
All channels are updated by polling regularly.
If you require a specific value to be refreshed, send a REFRESH command to it, that's what it is meant for.

Reply:
I've honestly never heard of the feature... and that should be a surprise to us both. 
For this specific binding one API response will update all channels at once. Effectively @pgfeller would have to implement some kind of caching to cope with REFRESH bursts. The dedicated refresh channel makes more sense to me. I'm undecided.

Reply:
> Effectively @pgfeller would have to implement some kind of caching

Correct - he will need that in any case, that's why the framework already provides the [ExpiringCache](https://www.eclipse.org/smarthome/documentation/features/frameworkUtilities.html#simple-expiring-and-reloading-cache).
Note that during startup and linked channel will receive a REFRESH command already, so this should be handled in any case.

Reply:
@kaikreuzer Can you point me to a full example of how this caching has to be implemented; I´ll then try to implement a proper refresh as well.

Reply:
Done.